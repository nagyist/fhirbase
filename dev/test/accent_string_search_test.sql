--db:fhirb -e
SET escape_string_warning=off;

--{{{
\set cfg '{"base":"https://test.me"}'
\set device `cat test/fixtures/device.json`

BEGIN;

  SELECT assert_eq (
    'Joe Acme',
    _unaccent_string('Jóe Ácme'),
    '_unaccent_string');

  WITH device AS (
    SELECT *
    FROM fhir_create(:'cfg', 'Device', :'device'::jsonb, '[]'::jsonb) as bundle
  ), searching AS (
    SELECT d.*,
      fhir_search(:'cfg', 'Device', 'model=Joe') as res,
      fhir_search(:'cfg', 'Device', 'manufacturer=Ácme') as param,
      fhir_search(:'cfg', 'Device', 'model=Jóe') as res_param
    FROM device d
  )
  SELECT
    assert_eq(
      d.bundle#>>'{entry,0,id}',
      d.res#>>'{entry,0,id}',
      'accent in resource'),
    assert_eq(
      d.bundle#>>'{entry,0,id}',
      d.param#>>'{entry,0,id}',
      'accent in search parameter'),
    assert_eq(
      d.bundle#>>'{entry,0,id}',
      d.res_param#>>'{entry,0,id}',
      'accent in resource and search parameter')
  FROM searching d;

ROLLBACK;
--}}}
