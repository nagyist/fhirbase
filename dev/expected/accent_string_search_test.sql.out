SET escape_string_warning=off;
SET
BEGIN;
BEGIN
SELECT assert_eq (
    'Joe Acme',
    _unaccent_string('Jóe Ácme'),
    '_unaccent_string');
WITH device AS (
    SELECT *
    FROM fhir_create('{"base":"https://test.me"}', 'Device', '{
  "resourceType": "Device",
  "type": {
    "coding": [
      {
        "system": "http://snomed.info/sct",
        "code": "86184003",
        "display": "Electrocardiographic monitor and recorder"
      }
    ],
    "text": "ECG"
  },
  "model": "Jóe",
  "manufacturer": "Acme"
}'::jsonb, '[]'::jsonb) as bundle
  ), searching AS (
    SELECT d.*,
      fhir_search('{"base":"https://test.me"}', 'Device', 'model=Joe') as res,
      fhir_search('{"base":"https://test.me"}', 'Device', 'manufacturer=Ácme') as param,
      fhir_search('{"base":"https://test.me"}', 'Device', 'model=Jóe') as res_param
    FROM device d
  )
  SELECT
    assert_eq(
      d.bundle#>>'{entry,0,id}',
      d.res#>>'{entry,0,id}',
      'accent in resource'),
    assert_eq(
      d.bundle#>>'{entry,0,id}',
      d.param#>>'{entry,0,id}',
      'accent in search parameter'),
    assert_eq(
      d.bundle#>>'{entry,0,id}',
      d.res_param#>>'{entry,0,id}',
      'accent in resource and search parameter')
  FROM searching d;
ROLLBACK;
ROLLBACK
